Deceive/MainController.cs

```diff
@@ -31,17 +33,34 @@ public MainController()
             trayIcon.ShowBalloonTip(5000);
             LoadStatus();
             SetupMenuItems();
+
+            //Update status when LCU launches
+            //Try every 5 seconds until LCU is ready to accept our status
+            initLCUStatusTimer.Tick += new EventHandler(InitLCUStatus);
+            initLCUStatusTimer.Interval = 5000;
+            initLCUStatusTimer.Start();
+        }
+
+        private void InitLCUStatus(object sender, EventArgs e)
+        {
+            try
+            {
+                Utils.SendStatusToLCU(status);
+                //stop Timer when changing LCU status finally worked and we get no WebException
+                initLCUStatusTimer.Dispose();
+            }
+            catch { /* LCU is not ready yet. */ }
```

Comment:
It's probably better to use an async function here instead of being hacky with the Timer. Something along the lines of:

```c#
private async void InitLCUStatus() {
	while (true) {
		try {
			Utils.SendStatusToLCU(status);
			return;
		} catch {
			// LCU is not ready yet. Wait for a bit.
			await Task.Delay(5000);
		}
	}
}
```

(and then just call that from the constructor without awaiting the result)

Reply:
I had something similar directly in the SendStatusToLCU function, didn't work out too well somehow. Your solution definitely looks a lot better though. Will try that!